libvast/vast/bloom_filter.hpp

```diff
@@ -110,7 +110,11 @@ class bloom_filter : detail::equality_comparable<
 
   template <class Inspector>
   friend auto inspect(Inspector& f, bloom_filter& x) {
-    return f(caf::meta::type_name("bloom_filter"), x.hasher_, x.bits_);
+    auto result = f(caf::meta::type_name("bloom_filter"), x.hasher_, x.bits_);
+    // When deserializing into a vector that was already bigger than required,
+    // CAF will reuse the storage but not release the excess afterwards.
+    x.bits_.shrink_to_fit();
```

Comment:
Would it make sense to put this mutation in the calling context, i.e., after deserializing? Since this code path is used for both serialization and deserialization, and potentially by other concepts as well, I'm not sure if it the right place.

Reply:
@dominiklohmann 's suggestion below is the best option here.

Reply:
Okay.